lib/queue.rb

```diff
@@ -1,31 +1,117 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @front = -1
+    @back = -1
+    @size = 20
+    # @empty = true
+    # @front = @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if ((@front == 0 && @back == @size - 1) || (@back == (( @front - 1) % ( @size - 1))))
```

Comment:
I think this looks a little simpler

```suggestion
    if ((@front == 0 && @back == @size - 1) || (@front == (( @back + 1) % ( @size))))
```